service/src/main/java/bio/terra/workspace/service/workspace/GcpCloudContextService.java

```diff
@@ -119,8 +120,8 @@ public GcpCloudContext getRequiredGcpCloudContext(
           samService.getWorkspacePolicy(workspaceUuid, WsmIamRole.READER, userRequest));
       context.setSamPolicyApplication(
           samService.getWorkspacePolicy(workspaceUuid, WsmIamRole.APPLICATION, userRequest));
+      workspaceDao.updateCloudContext(workspaceUuid, CloudPlatform.GCP, context.serialize());
     }
-    workspaceDao.updateCloudContext(workspaceUuid, CloudPlatform.GCP, context.serialize());
```

Comment:
We should only be updating the cloud context if we're modifying it. Otherwise, we're just reading a DB value -> not modifying it -> writing it back to the DB